homeassistant/components/edl21/sensor.py

```diff
@@ -435,4 +450,17 @@ def extra_state_attributes(self):
     @property
     def native_unit_of_measurement(self):
         """Return the unit of measurement."""
-        return self._telegram.get("unit")
+        unit = self._telegram.get("unit")
+
+        if unit is None:
+            return None
+
+        if not (mapped_unit := SENSOR_UNIT_MAPPING.get(unit)):
+            _LOGGER.error(
```

Comment:
We don't want side effects like logging in entity state properties. I think we should just blow up if the unit is incorrect.

Reply:
By "blowing up" you mean raising an exception? Or returning null?

Reply:
Yeah, raise an exception. KeyError will be raised by default if the key is missing in the unit map.